src/regular-expression-comment.ts

```diff
@@ -0,0 +1,42 @@
+// regular-expression-comment.ts
+
+/*
+ * Copyright (c) 2021-2023 Check Digit, LLC
+ *
+ * This code is licensed under the MIT license (see LICENSE.txt for details).
+ */
+
+import type { Rule } from 'eslint';
+
+export default {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Require comments for regular expressions before or on the same line',
+      url: 'https://github.com/checkdigit/eslint-plugin',
+    },
+  },
+  create(context) {
+    return {
+      Literal(node) {
+        if (node.value instanceof RegExp) {
+          const regexLine = node.loc?.start.line;
+          if (regexLine) {
+            const previousLine = regexLine - 1;
+            const hasRegexComment =
+              context.sourceCode.getAllComments().some((comment) => comment.loc?.start.line === previousLine) ||
```

Comment:
this is a little inefficient, to start with we only should scan all comments once (e.g. `... .some((comment) => comment.loc?.start.line === previousLine || comment.loc?.start.line === regexLine)`).  even then seems excessive, but I suppose there aren't normally a ton of regexs in our code.  Is there a more efficient way to see if there is a comment on this line or the previous, other than scanning a list of all comments in the code?

Reply:
Sure  we do have something like getCommentsBefore(node) that would check before this line with regex similarly getCommentsInside(node). But it somehow doesn't return the comments even though they exist, so have to rewrite this way.